engine/src/main/java/io/camunda/zeebe/engine/state/migration/to_8_4/DbColumnFamilyCorrectionMigrationState.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.engine.state.migration.to_8_4;
+
+import io.camunda.zeebe.db.ColumnFamily;
+import io.camunda.zeebe.db.TransactionContext;
+import io.camunda.zeebe.db.ZeebeDb;
+import io.camunda.zeebe.db.impl.DbBytes;
+import io.camunda.zeebe.db.impl.DbCompositeKey;
+import io.camunda.zeebe.db.impl.DbLong;
+import io.camunda.zeebe.db.impl.DbString;
+import io.camunda.zeebe.engine.state.migration.MigrationTaskState;
+import io.camunda.zeebe.engine.state.migration.MigrationTaskState.State;
+import io.camunda.zeebe.protocol.ZbColumnFamilies;
+
+public class DbColumnFamilyCorrectionMigrationState {
+
+  // (signalName, subscriptionKey => SignalSubscription)
+  //  private final DbCompositeKey<DbString, DbLong> signalNameAndSubscriptionKey;
+  //  private final ColumnFamily<byte[], SignalSubscription>
+  //      signalNameAndSubscriptionKeyColumnFamily;
+
+  private final ColumnFamily<DbBytes, DbBytes> signalNameAndSubscriptionKeyColumnFamily;
+
+  private final ColumnFamily<DbString, MigrationTaskState> migrationStateColumnFamily;
+  private final DbString migrationIdentifier;
+  private final MigrationTaskState migrationTaskState;
+
+  public DbColumnFamilyCorrectionMigrationState(
+      final ZeebeDb<ZbColumnFamilies> zeebeDb, final TransactionContext transactionContext) {
+    signalNameAndSubscriptionKeyColumnFamily =
+        zeebeDb.createColumnFamily(
+            ZbColumnFamilies.DEPRECATED_SIGNAL_SUBSCRIPTION_BY_NAME_AND_KEY,
+            transactionContext,
+            new DbBytes(),
+            new DbBytes());
+
+    migrationIdentifier = new DbString();
+    migrationTaskState = new MigrationTaskState();
+    migrationStateColumnFamily =
+        zeebeDb.createColumnFamily(
+            ZbColumnFamilies.MIGRATIONS_STATE,
+            transactionContext,
+            migrationIdentifier,
+            migrationTaskState);
+  }
+
+  public void correctColumnFamilyPrefix() {
+    signalNameAndSubscriptionKeyColumnFamily.forEach(
+        (key, value) -> {
+          final DbString signalName = new DbString();
+          final DbLong subscriptionKey = new DbLong();
+          final DbCompositeKey<DbString, DbLong> signalNameAndSubscriptionKey =
+              new DbCompositeKey<>(signalName, subscriptionKey);
+          try {
+            signalNameAndSubscriptionKey.wrap(key.getDirectBuffer(), 0, key.getLength());
+          } catch (final Exception e) {
```

Comment:
Just small input: I'm not 100% sure whether this is sufficient enough. But of course I also don't know what you have planned for the other types of CFs. It might be worth to check the length, and fields which have been written in the key. 

Why I'm unsure?

Because the composite keys are not writing the length itself and not what they types to be expected https://github.com/camunda/zeebe/blob/main/zb-db/src/main/java/io/camunda/zeebe/db/impl/DbCompositeKey.java#L56-L60

Let's say you have a [String](https://github.com/camunda/zeebe/blob/main/zb-db/src/main/java/io/camunda/zeebe/db/impl/DbString.java#L48-L54) and [Long](https://github.com/camunda/zeebe/blob/main/zb-db/src/main/java/io/camunda/zeebe/db/impl/DbLong.java#L37) as composite. 

The key would look like this: <CF, <STR_LEN, STRING, LONG>>

If I have a long long

<CF, <LONG, LONG>>

One problem we would have if the string len is zero then it is exact the same as the above. This means it would not fail to parse.

If I expect a <CF, <STR_LEN, String>>

I would still be able to parse the first composite key <CF, <STR_LEN, STRING, LONG>>, etc.

Not sure whether we have more examples. But I think you get the idea.